node/pkg/chain/tests/noncemanager_test.go

```diff
@@ -0,0 +1,65 @@
+//nolint:all
+package tests
+
+import (
+	"sync"
+	"testing"
+
+	"bisonai.com/orakl/node/pkg/chain/noncemanager"
+)
+
+func TestNonceManager(t *testing.T) {
+	nm := noncemanager.New()
+	address := "0x123"
+
+	nm.SetNonce(address, 1)
+	nonce, err := nm.GetNonce(address)
+	if err != nil {
+		t.Fatalf("Expected no error, got %v", err)
+	}
+	if nonce != 1 {
+		t.Fatalf("Expected nonce to be 1, got %d", nonce)
+	}
+
+	newNonce, err := nm.GetNonceAndIncrement(address)
+	if err != nil {
+		t.Fatalf("Expected no error, got %v", err)
+	}
+	if newNonce != 1 {
+		t.Fatalf("Expected new nonce to be 2, got %d", newNonce)
```

Comment:
expected to be 1?

Reply:
yh the text inside t.Fatalf should be updated